ui/src/app/pages/signed-in/signed-in.tsx

```diff
@@ -127,41 +174,7 @@ export const SignedInImpl = (spinnerProps: WithSpinnerOverlayProps) => {
     checkStoresLoaded();
   }, [profileState, tiers]);
 
-  // DEMOGRAPHIC_SURVEY_SESSION_KEY is set in session when the user selects Maybe Later Button on
-  // Demographic Survey Page and is cleared out on signOut.
-  // So, if this key exist, it means user should not be redirected to demographic survey page.
-  const hasDismissedDemographicSurvey = sessionStorage.getItem(
-    DEMOGRAPHIC_SURVEY_SESSION_KEY
-  );
-
-  const shouldRedirectToDemographicSurveyPage = () => {
-    const { demographicSurveyV2 } = profileState.profile;
-    return (
-      shouldShowDemographicSurvey(profileState.profile) &&
-      !demographicSurveyV2 &&
-      !hasDismissedDemographicSurvey
-    );
-  };
-
-  const SignedInContent = () =>
```

Comment:
Defining components in render methods can cause problems.  Some discussion about why: https://stackoverflow.com/questions/52372040/is-it-an-anti-pattern-to-define-a-function-component-inside-the-render-functio

Reply:
There's an eslint rule that can help with this: https://github.com/all-of-us/workbench/pull/9152

I may look at applying it throughout the RWB.